src/core/thread/mle.cpp

```diff
@@ -81,7 +81,10 @@ Mle::Mle(Instance &aInstance)
     , mAttachTimer(aInstance, Mle::HandleAttachTimer)
     , mDelayedResponseTimer(aInstance, Mle::HandleDelayedResponseTimer)
     , mMessageTransmissionTimer(aInstance, Mle::HandleMessageTransmissionTimer)
+    , mDetachGracefullyTimer(aInstance, Mle::HandleDetachGracefullyTimer)
```

Comment:
Not in this PR, but I think we may consider reusing `AttachTimer` for detach.

Reply:
SGTM. Detaching should interrupt any ongoing attaching process, so reusing `AttachTimer` is good. We can do that in future PRs.